EditImprovements/Edit.patch

```diff
@@ -0,0 +1,285 @@
++              name[i] := "."; name[i+1] := "M"; name[i+2] := "o"; name[i+3] := "d"; name[i+4] := 0X;
++              Texts.OpenScanner(S, V.dsc(TextFrames.Frame).text, 0);
++              Texts.Scan(S);
++              IF (S.class = Texts.Name) & (S.s # name) THEN
++                Oberon.AllocateUserViewer(V.X, X, Y);
++                V := MenuViewers.New(
++                  TextFrames.NewMenu(name, StandardMenu),
++                  TextFrames.NewText(TextFrames.Text(name), 0),
++                  TextFrames.menuH, X, Y)
++              END
++            END
++          END;
++          F := V.dsc.next(TextFrames.Frame);
++          TextFrames.RemoveSelection(F);
++          TextFrames.RemoveCaret(F);
++          Oberon.RemoveMarks(F.X, F.Y, F.W, F.H);
++          ShowPos(F, pos);
++          Oberon.PassFocus(V);
++          TextFrames.SetCaret(F, pos)
++        END
++      END
++    END
++  END Locate;
++
+   PROCEDURE Search*;  (*uses global variables M, pat, d for Boyer-Moore search*)
+-    VAR Text: TextFrames.Frame;
++    VAR F: TextFrames.Frame;
+       V: Viewers.Viewer;
+       R: Texts.Reader;
+       T: Texts.Text;
+-      pos, beg, end, prevTime, len: LONGINT; n, i, j: INTEGER;
++      pos, beg, end, prevTime, len, n, i, j: INTEGER; ch: CHAR;
+       buf: ARRAY 32 OF CHAR;
+ 
+     PROCEDURE Forward(n: INTEGER; VAR R: Texts.Reader; VAR buf: ARRAY OF CHAR);
+@@ -141,7 +217,7 @@
+   BEGIN V := Oberon.Par.vwr;
+     IF Oberon.Par.frame # V.dsc THEN V := Oberon.FocusViewer END;
+     IF (V.dsc # NIL) & (V.dsc.next IS TextFrames.Frame) THEN
+-      Text := V.dsc.next(TextFrames.Frame);
++      F := V.dsc.next(TextFrames.Frame);
+       prevTime := time; Oberon.GetSelection(T, beg, end, time);
+       IF time > prevTime THEN
+         Texts.OpenReader(R, T, beg);
+@@ -153,10 +229,10 @@
+         j := 0;
+         WHILE j # M - 1 DO d[ORD(pat[j])] := M - 1 - j; INC(j) END
+       END;
+-      IF Text.hasCar THEN pos := Text.carloc.pos ELSE pos := 0 END;
+-      len := Text.text.len;
+-      Texts.OpenReader(R, Text.text, pos);
+-      Forward(M, R, buf); pos := pos + M;
++      IF F.hasCar THEN pos := F.carloc.pos ELSE pos := 0 END;
++      len := F.text.len;
++      Texts.OpenReader(R, F.text, pos);
++      Forward(M, R, buf); INC(pos, M);
+       j := M;
+       REPEAT DEC(j) UNTIL (j < 0) OR (buf[j] # pat[j]);
+       WHILE (j >= 0) & (pos < len) DO
+@@ -164,69 +240,14 @@
+         REPEAT DEC(j) UNTIL (j < 0) OR (buf[j] # pat[j])
+       END ;
+       IF j < 0 THEN
+-        TextFrames.RemoveSelection(Text); TextFrames.RemoveCaret(Text);
+-        Oberon.RemoveMarks(Text.X, Text.Y, Text.W, Text.H);
+-        TextFrames.Show(Text, pos - 300); Oberon.PassFocus(V);
+-        TextFrames.SetCaret(Text, pos)
++        TextFrames.RemoveSelection(F); TextFrames.RemoveCaret(F);
++        Oberon.RemoveMarks(F.X, F.Y, F.W, F.H);
++        ShowPos(F, pos);
++        Oberon.PassFocus(V);
++        TextFrames.SetCaret(F, pos)
```

Comment:
variable rename only?

Reply:
Yup, I mentioned it in the correspondence that F uppercase is the name for variables of type TextFrames.Frame throughout the whole Oberon System for ages. And even the change from F to text made by someone was not carried through completely.

Reply:
If you believe this change belongs to this modification, please mention it in the README.